include/rmm/mr/default_memory_resource.hpp

```diff
@@ -51,5 +51,70 @@ device_memory_resource* get_default_resource();
 device_memory_resource* set_default_resource(
     device_memory_resource* new_resource);
 
+namespace detail{
+
+
+/**---------------------------------------------------------------------------*
+ * @brief gets the default memory_resource when none is set
+ *
+ * A static function which will return a singleton cuda_memory_resource
+ *
+ *
+ * @return device_memory_resource* a pointer to the static
+ * cuda_memory_resource
+ *---------------------------------------------------------------------------**/
+device_memory_resource* initial_resource();
+
+/**---------------------------------------------------------------------------*
+ * @brief gets a cuda_memory_resource
+ *
+ * A static function which will return a singleton cuda_memory_resource
+ *
+ *
+ *
+ * @return device_memory_resource* a pointer to the static
+ * cuda_memory_resource
+ *---------------------------------------------------------------------------**/
+device_memory_resource* cuda_resource();
+
+/**---------------------------------------------------------------------------*
+ * @brief gets a cnmem_memory_resource
+ *
+ * A static function which will return a singleton cnmem_memory_resource which
+ * manages a pool
+ *
+ *
+ * @param pool_size The initial size of the pool
+ * @return device_memory_resource* a pointer to the static
+ * cnmem_memory_resource
+ *---------------------------------------------------------------------------**/
+device_memory_resource* pool_resource(std::size_t pool_size = 0);
+
+/**---------------------------------------------------------------------------*
+ * @brief gets a cnmem_managed_memory_resource
+ *
+ * A static function which will return a singleton cnmem_memory_resource which
+ * manages a pool of UVM memory
+ *
+ *
+ * @param pool_size The initial size of the pool
+ * @return device_memory_resource* a pointer to the static
+ * cnmem_managed_memory_resource
+ *---------------------------------------------------------------------------**/
+device_memory_resource* managed_pool_resource(std::size_t pool_size = 0);
```

Comment:
This doesn't have a definition anywhere.

Reply:
i added it I hadn't saved it